lib/nxdk/xbe.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2019 Stefan Schmidt
+ *
+ * Licensed under the MIT License
+ */
+
+#ifndef __NXDK_XBE_H__
+#define __NXDK_XBE_H__
+
+#ifdef __cplusplus
+{
+#endif
+
+#ifndef __cplusplus
+#include <stdbool.h>
+#endif
+
+void XGetXbePath (char *path, bool includeFileName);
```

Comment:
Does this return a path like `D:/default.xbe` or `\Device\\Harddisk0\\Partition2\\default.xbe`?

Reply:
Like `\Device\CdRom0\default.xbe`.

Reply:
We should document this, as most people would probably expect a mounted path instead.

I'm always concerned about the future-proofness, as I want to avoid breakage of nxdk apps if we have to change APIs.

If we add more path-related functions, we might want to make them return the same path style, too. If it doesn't make sense (it might make sense to address some paths using drive letters for example).
Maybe we should name this `XGetXbeDevicePath` to clarify this / avoid confusion in the future?

Also, would these paths also work for our libc `fopen` etc.?

Reply:
Following our Discord discussion, this is now named `XGetCurrentXbeNtPath`. I added Doxygen-style documentation as well.

This path style is not compatible with winapi and libc.